### ISSUE TITLE
Interpretation of 2.2.1 "Timing Adjustable" for content that counts down (and resets) time

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#908]
We are wondering about the WG's interpretation of 2.2.1 "Timing Adjustable" regarding web application content that counts down (and resets) the time before logout. We typically see, for exampe, a count-down of 15 min in online banking contexts, or 1 hr for db applications (somewhere near the top of the page).  Each user action resets the time before logout to the default value. There is usually no way of 1. turning off; 2. Adjusting (in the sense of setting a control); or 3. Extending (after a warning via dialog or alert), and the Essential exception doesn't seem to apply.
A strict interpretation would then suggest that implementations that automatically reset the time and display the count-down, even when the interval may be as long a 1 hour, would not be sufficient to pass, and this content would thus fail 2.1.1. What do people think? There is no positive Technique that would cover this scenario, but also no fitting Failure for it.

**Comment by mraccess77** [comment]
I generally consider this a pass as you can extend the time.

**Comment by detlevhfischer** [comment]
If others agree with @mraccess77, it would be quite helpful to explain in the understanding text (or even via errata in the normative text) that 'adjust' can also mean 'interact with content so that timeout is automatically reset'. Should we write a General Technique that covers this very frequent scenario to remove any ambiguity?

**Comment by patrickhlauke** [comment]
>  'adjust' can also mean 'interact with content so that timeout is automatically reset'

i believe it's more that *'extend'* can also mean... (not adjust)

**Comment by jake-abma** [comment]
> We typically see, for example, a count-down of 15 min in online banking contexts, or 1 hr for db applications (somewhere near the top of the page)

> Each user action resets the time before logout to the default value.



Each user action resets the time before logout to the default value.

Extend
The user is warned before time expires and given at least 20 seconds to extend the time limit with a simple action (for example, "press the space bar"), and the user is allowed to extend the time limit at least ten times; or

**Comment by jake-abma** [comment]
> We typically see, for example, a count-down of 15 min in online banking contexts, or 1 hr for db applications (somewhere near the top of the page)

So a consistent (time) indicator always present, in contrary to for example, a dialog 20 second before time expires (which is a clear warning!)

> Each user action resets the time before logout to the default value.

Indeed as Patrick mentioned this is extending

> Extend
> The user is warned before time expires and given at least 20 seconds to extend the time limit with a simple action (for example, "press the space bar"), and the user is allowed to extend the time limit at least ten times; or

The question I have here is: "does the counter in the right corner 'counts' as a warning? And is this clear enough? What indicator is there that you'll be logged out?"

The text is not strong enough right now to cover this in a clear manner.

**Comment by jake-abma** [comment]
> We typically see, for example, a count-down of 15 min in online banking contexts, or 1 hr for db applications (somewhere near the top of the page)

So a consistent (time) indicator always present, in contrary to for example, a dialog 20 second before time expires (which is a clear warning!)

> Each user action resets the time before logout to the default value.

Indeed as Patrick mentioned this is extending

> Extend
> The user is warned before time expires and given at least 20 seconds to extend the time limit with a simple action (for example, "press the space bar"), and the user is allowed to extend the time limit at least ten times; or

The question I have here is: "does the counter in the right corner 'counts' as a warning? And is this clear enough? What indicator is there that you'll be logged out?"

The text is not strong enough right now to cover this in a clear manner.

**Comment by detlevhfischer** [comment]
The way “extend” is phrased first the morning and then 20 seconds to react, doesn’t really sound like the automatic extension through any action. I personally would not have thought that a permanently visible time indication in the corner of the screen a warning - The 20 sec time to extend suggest a dialogue or alert.
If the working group thinks that the case I have described passes, then I think it would be useful to make that clear in the understanding text and also to have a positive technique for that.

**Comment by bruce-usab** [comment]
I agree w/ @mraccess77 that the case @detlevhfischer has described passes, and that it would be useful to make that clear in the understanding text and also to have a positive technique for that.

Just to expand on my reasoning, “the user is warned before time expires” could be instructions provided at the beginning of the session (no count-down timer required), and the time limit resetting from natural interaction with the DB is “a simple action”.

I agree that the Extend phrasing implies an explicit warning and confirmation dialog box, but I never thought that was the only pattern case allowed for by the Extend clause.
